File: TransCoder_tokenized_test_set_functions/java/GENERATE_BINARY_PERMUTATIONS_1S_0S_EVERY_POINT_PERMUTATIONS.java
static void generate ( int ones, int zeroes, String str, int len ) {
  if ( len == str . length ( ) ) {
    System . out . print ( str + " " );
    return;
  }
  generate ( ones + 1, zeroes, str + "1", len );
  if ( ones > zeroes ) {
    generate ( ones, zeroes + 1, str + "0", len );
  }
}

File: TransCoder_tokenized_test_set_functions/python/GENERATE_BINARY_PERMUTATIONS_1S_0S_EVERY_POINT_PERMUTATIONS.py
def generate ( ones, zeroes, str, len1 ) :
    if ( len1 == len ( str ) ) :
        print ( str, end = " " )
        return
    generate ( ones + 1, zeroes, str + "1", len1 )
    if ( ones > zeroes ) :
        generate ( ones, zeroes + 1, str + "0", len1 )
    
